From: Peter Zijlstra <a.p.zijlstra@chello.nl>
Subject: perf, powerpc: Fix build borkage
Date: Fri Jul 1 15:24:53 2011
References: bnc#789200, fate#313753, fate#312066
Patch-mainline: v3.1-rc1
Git-commit: 29dfc4fd7e792854b30d771d4e299fbdd2adb763
Signed-off-by: Tony Jones <tonyj@suse.de>

    perf, powerpc: Fix build borkage
    
    The patch a8b0ca17b80e ("perf: Remove the nmi parameter from the swevent
    and overflow interface") missed a spot in the ppc hw_breakpoint code,
    fix this up so things compile again.
    
    Reported-by: Ingo Molnar <mingo@elte.hu>
    Cc: Anton Blanchard <anton@samba.org>
    Cc: Eric B Munson <emunson@mgebm.net>
    Signed-off-by: Peter Zijlstra <a.p.zijlstra@chello.nl>
    Link: http://lkml.kernel.org/n/tip-09pfip95g88s70iwkxu6nnbt@git.kernel.org
    Signed-off-by: Ingo Molnar <mingo@elte.hu>

diff --git a/arch/powerpc/include/asm/hw_breakpoint.h b/arch/powerpc/include/asm/hw_breakpoint.h
index 1c33ec1..80fd4d2 100644
--- a/arch/powerpc/include/asm/hw_breakpoint.h
+++ b/arch/powerpc/include/asm/hw_breakpoint.h
@@ -57,7 +57,7 @@ void hw_breakpoint_pmu_read(struct perf_event *bp);
 extern void flush_ptrace_hw_breakpoint(struct task_struct *tsk);
 
 extern struct pmu perf_ops_bp;
-extern void ptrace_triggered(struct perf_event *bp, int nmi,
+extern void ptrace_triggered(struct perf_event *bp,
 			struct perf_sample_data *data, struct pt_regs *regs);
 static inline void hw_breakpoint_disable(void)
 {
